spec/models/user_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+describe User, type: :model do
+  fixtures :users
+  subject { users(:one) } 
+  # subject { User.new }
+  # let(:valid_user) { users(:one) }
+
+  describe 'Validations' do
+    describe '#firstname' do
+      it 'is invalid when blank' do
+        subject.firstname = ' ' * 6
+        expect(subject).to_not be_valid
+      end
+
+      it 'is invalid when longer than 50 characters' do
+        subject.firstname = 'a' * 51
+        expect(subject).to_not be_valid
+      end
+
+      it 'is valid with proper data' do
+        expect(subject).to be_valid
+      end
+    end
+
+    describe '#lastname' do
+      it 'is invalid when blank' do
+        subject.lastname = ' ' * 6
+        expect(subject).to_not be_valid
+      end
+
+      it 'is invalid when longer than 50 characters' do
+        subject.lastname = 'a' * 51
+        expect(subject).to_not be_valid
+      end
+
+      it 'is valid with proper data' do
+        expect(users(:one)).to be_valid
+      end
+    end
+
+    describe '#bio' do
+      it 'is invalid when longer than 300 characters' do
+        subject.bio = 'a' * 301
+        expect(subject).to_not be_valid
+      end
+
+      it 'is valid with proper data' do
+        expect(users(:one)).to be_valid
+      end
+    end
+  end
+
+  describe 'Associations' do
+    it 'has many posts' do
+      assc = described_class.reflect_on_association(:posts)
+      expect(assc.macro).to eq :has_many
+    end
+
+    it 'has many notifications' do
+      assc = described_class.reflect_on_association(:notifications)
+      expect(assc.macro).to eq :has_many
+    end
+
+    it 'has many friends' do
```

Comment:
Actually `Associations` spec do not have much value.
You just test if `has_many :friends` was used in code 🤔 

And if you check it in rails c:

```
irb(main):002:0> u.friends
NameError: uninitialized constant User::Friend
	from (irb):2
irb(main):003:0> 
```

If you really want to test associations, it would be better to:

```
 it 'has many posts' do
    expect(subject.posts).not_to raise_error
 end
```

or even more precise:

```
 it 'has many posts' do
  user_post = Post.create(user_id: subject.id)
  expect(subject.reload.posts,first.id).to eq(user_post.id)
 end
```